modules/core/CMakeLists.txt

```diff
@@ -131,6 +131,9 @@ elseif(HAVE_CXX11 OR DEFINED OPENCV_ALLOCATOR_STATS_COUNTER_TYPE)
   endif()
 endif()
 
+if (CV_SEMIHOSTING)
+  add_definitions("-DCV_SEMIHOSTING")
+endif(CV_SEMIHOSTING)
```

Comment:
- Prefix `CV_` => `OPENCV_`
- no space after if: `if(...)` (it is CMake not C++)
- reuse `OPENCV_DISABLE_THREAD_SUPPORT` approach (probably as `PRIVATE`)